app/controllers/api/connect/v3/systems/systems_controller.rb

```diff
@@ -3,10 +3,12 @@ class Api::Connect::V3::Systems::SystemsController < Api::Connect::BaseControlle
   before_action :authenticate_system
 
   def update
-    # FIXME: need hwinfo -- require_params([:hwinfo])
-
     @system.hostname = params[:hostname] || _('Not provided')
-    # FIXME: @system.hw_info.update_attributes(hwinfo_params)
+    if @system.hw_info.present?
+      @system.hw_info.update(hwinfo_params)
+    else
+      @system.hw_info_attributes = hwinfo_params
```

Comment:
I think this is where the issues of `hw_info_attributes` comes out.

If it's not present: `@system.hw_info.update()`
otherwise:`@system.hw_info_attributes =`

To me, it seems like you are updating two different things.

Also, consider if this means that this controller knows too much.

Reply:
the alternative is to use `find_or_initialize_by`. maybe it's better